mlflow/models/evaluation/evaluators/shap.py

```diff
@@ -239,7 +239,13 @@ def _adjust_axis_tick():
             pyplot.yticks(fontsize=10)
 
         def plot_beeswarm():
-            shap.plots.beeswarm(shap_values, show=False, color_bar=True)
+            shap.plots.beeswarm(
+                shap_values,
+                show=False,
+                color_bar=True,
+                # Workaround for https://github.com/shap/shap/issues/3901
+                color="cool" if Version(np.__version__).major >= 2 else None,
```

Comment:
A string needs to be used to avoid https://github.com/shap/shap/issues/3901.

Reply:
Considered wrapping `shap.plots.beeswarm` with try-catch to swallow the error, but a plot with a different color map is better than no plot.

Reply:
I think shap will continue to cause issues like this one. We might want to turn if off by default.

Reply:
I think this fix is fine. 
I also agree that we should consider making the plot generation aspects of shap and possible shap in general opt-in only (this isn't the first time that these plots have created issues and have resulted in broken APIs for users)